build.sbt

```diff
@@ -1,31 +1,61 @@
-crossScalaVersions := Seq("2.13.6", "2.12.14")
-scalaVersion := crossScalaVersions.value.head
-
-name := "library-example"
+import xerial.sbt.Sonatype._
 
-// also used as a `groupId` by Sonatype
-organization := "ch.epfl.scala"
+lazy val `library-example` = project
+  .in(file("."))
+  .settings(
+    name := "library-example",
+    organization := "ch.epfl.scala",
+    description := "A library that does nothing useful",
+    crossScalaVersions := Seq("3.0.0", "2.13.6", "2.12.14"),
+    scalaVersion := crossScalaVersions.value.head,
+    libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test,
+    testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+  ).settings(publishSettings)
+   .settings(siteSettings)
+   .enablePlugins(MdocPlugin)
 
-libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val root = project
+  .in(file("."))
+  .aggregate(`library-example`)
 
-import xerial.sbt.Sonatype._
-sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch"))
-// indicate the open source licenses that apply to our project
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-// publish to the Sonatype repository
-publishTo := sonatypePublishToBundle.value
+lazy val siteSettings = {
+  /*
+  Combine scaladoc and mdoc.
+  To do so, 
+   1. copy site to target/processed_site
+   2. use mdoc to process site/docs/ **.md
+   3. mdoc outputs new md files in target/processed_site/docs
```

Comment:
Will `processed_site` contain both the files in `site`, and the output of mdoc?

Reply:
In [the deployed branch](https://github.com/vincenzobaz/library-example/tree/gh-pages/docs) there are no duplicated files and code is evaluated by mdoc (for example [here](https://github.com/vincenzobaz/library-example/blob/c783e358549c7d4e94a9276d40636fe7fa1c5868/docs/getting-started.html#L20)).

I suppose that mdoc overwrites files with the same name

Reply:
Does mdoc copy all the files anyway? Do we need to copy anything manually, then?

Reply:
We need to copy the rest of the files required for building the site, including the `sidebar.yml` and all the html templates.

Reply:
OK, so IIUC, we manually copy all the source files, then mdoc overwrites the .md files?

Reply:
Correct.